src/service.ts

```diff
@@ -0,0 +1,250 @@
+    "wallet.vote",
+    "wallet.unvote",
+    "round.created",
+    "activedelegateschanged",
+];
+
+const CUSTOM_EVENTS = ["activedelegateschanged"];
+const CUSTOM_EVENT_MAPPING = {
+    activedelegateschanged: AppEnums.BlockEvent.Forged, // AppEnums.RoundEvent.Created,
+};
+
+let LAST_ACTIVE_DELEGATES_CACHED: string[] = [];
+
+@Container.injectable()
+export default class Service {
+    @Container.inject(Container.Identifiers.EventDispatcherService)
+    private readonly emitter!: Contracts.Kernel.EventDispatcher;
+
+    @Container.inject(Container.Identifiers.LogService)
+    private readonly logger!: Contracts.Kernel.Logger;
+
+    @Container.inject(Container.Identifiers.TriggerService)
+    private readonly triggers!: Services.Triggers.Triggers;
+
+    // @Container.inject(Container.Identifiers.DposState)
+    // @Container.tagged("state", "clone")
+    // private readonly dposState!: Contracts.State.DposState;
+
+    @Container.inject(Container.Identifiers.WalletRepository)
+    // why state, blockchain - why not state, clone?
+    @Container.tagged("state", "blockchain")
+    private readonly walletRepository!: Contracts.State.WalletRepository;
+
+    private events = {};
+
+    public async listen(options: IOptions): Promise<void> {
+        LAST_ACTIVE_DELEGATES_CACHED = await this.getActiveDelegates();
+
+        for (const webhook of options.webhooks) {
+            for (const event of webhook.events) {
+                if (!VALID_EVENTS.includes(event)) {
+                    this.logger.warning(
+                        `[deadlock-delegate/notifier] ${event} is not a valid event. Check events in your deadlock-notifier configuration`,
+                    );
+                    continue;
+                }
+
+                if (!this.events[event]) {
+                    this.events[event] = [];
+                }
+
+                this.events[event].push({
+                    endpoint: webhook.endpoint,
+                    payload: webhook.payload,
+                });
+            }
+        }
+        Object.keys(this.events).forEach((event) => this.subscribe(event));
+    }
+
+    private async getActiveDelegates() {
+        // im not sure why I need to use triggers, but trying to inject dposState and calling
+        // getActiveDelegates method from there didn't work
+        const activeDelegates: Contracts.State.Wallet[] | undefined = await this.triggers.call(
+            "getActiveDelegates",
+            {},
+        );
+        if (!activeDelegates) return [];
+        return activeDelegates.map((wallet) => wallet.getAttribute("delegate.username"));
+    }
+
+    private subscribe(event: string) {
+        let customEventName: string | undefined = undefined;
+        if (CUSTOM_EVENTS.includes(event)) {
+            customEventName = event;
+            event = CUSTOM_EVENT_MAPPING[event];
+        }
+
+        // for some yet unknown reason, if handlers are defined at the class level, we can not access
```

Comment:
This is caused by how JavaScript handles context of instances, not a behaviour specific to core.